### PR TITLE
Implement init argument for string.gmatch

### PR DESCRIPTION
**Pull request by arnodel**
Introduced in Lua 5.4

### CODE REVIEW

**Comment by cderici** [comment]
It would be very helpful if you can in the PR description explain what the change/feature is, or give a small example, or provide a link to a lua doc that explains what's being introduced/changed. Thanks!

File: lib/stringlib/lua/matching.lua [hunk 1]
```diff
@@ -120,6 +123,35 @@ do
...
+
+    -- Lua 5.4 introduces an optional third argument to string.gmatch,
+    -- specifying where to start the search.
+
+    for w in string.gmatch(s, "%a+", 8) do
+        print(w)
+    end
+    --> =orld
+    --> =from
+    --> =Lua
```
Comment (cderici) [inline lib/stringlib/lua/matching.lua:135]: The results of these tests seem wrong to me: 

I think this one should produce:
```
rld
to
Lua
```

File: lib/stringlib/lua/matching.lua [hunk 2]
```diff
@@ -120,6 +123,35 @@ do
...
+    for w in string.gmatch(s, "%a+", -1) do
+        print(w)
+    end
+    --> =a
+
+    for w in string.gmatch(s, "%a+", -7) do
+        print(w)
+    end
+    --> =rom
+    --> =Lua
```
Comment (cderici) [inline lib/stringlib/lua/matching.lua:146]: and this one:
```
to
Lua
```

File: lib/stringlib/lua/matching.lua [hunk 3]
```diff
@@ -120,6 +123,35 @@ do
...
+   
+    for w in string.gmatch(s, "%a+", -100) do
+        print(w)
+    end
+    --> =hello
+    --> =world
+    --> =from
+    --> =Lua
```
Comment (cderici) [inline lib/stringlib/lua/matching.lua:154]: and this one:
```
from
world
to
Lua
```

am I missing something?

**Review by cderici — CHANGES_REQUESTED** [review]

**Comment by cderici** [comment]
I checked the results with [https://www.lua.org/cgi-bin/demo](https://www.lua.org/cgi-bin/demo)

File: lib/stringlib/lua/matching.lua (hunk 1 above)
Comment (arnodel) [reply lib/stringlib/lua/matching.lua:135]: `s` is the string "hello world from Lua", so I think my test is correct (same for the other remarks). What do you think `s` is?

Comment (cderici) [reply lib/stringlib/lua/matching.lua:135]: Ah that makes sense, in the diff I saw a line `local s = "from=world, to=Lua"` right above these tests, thought that was the `s` that's used, my bad :+1:

**Review by cderici — APPROVED** [review]

**Comment by arnodel** [comment]
> It would be very helpful if you can in the PR description explain what the change/feature is, or give a small example, or provide a link to a lua doc that explains what's being introduced/changed. Thanks!

Sorry - I try to do that usually but forgot this time!

**Pull request merged**